api/src/test/java/org/openmrs/api/EncounterServiceTest.java

```diff
@@ -425,6 +428,33 @@ public void saveEncounter_shouldCascadeSaveToContainedConditions() {
 		assertTrue(savedConditions.contains(pregnancy));
 		assertTrue(savedConditions.contains(edema));
 	}
+
+	@Test
+	public void saveEncounter_shouldCascadeSaveToContainedDiagnoses() {
+		// setup
+		Encounter encounter = buildEncounter();
+		Diagnosis diagnosis1 = new Diagnosis();
+		diagnosis1.setDiagnosis(new CodedOrFreeText(null, null, "Fever"));
+		diagnosis1.setRank(1);
+		diagnosis1.setCertainty(ConditionVerificationStatus.CONFIRMED);
+
+		Diagnosis diagnosis2 = new Diagnosis();
+		diagnosis2.setDiagnosis(new CodedOrFreeText(null, null, "Also fever"));
+		diagnosis2.setRank(2);
+		diagnosis2.setCertainty(ConditionVerificationStatus.PROVISIONAL);
+		
+		// replay
+		Set<Diagnosis> diagnoses = new HashSet<>();
```

Comment:
What are you using this hashset for?

Reply:
Nothing, some leftovers from an earlier code.